### ISSUE TITLE
Minimum types proposal

### ISSUE THREAD

**Issue by crgimenes** [gasgo/gasgo#6]
Initial proposal for a minimum list of supported types.

- int64
- unix epoch (same as int64 with some additional functions)
- string
- record (contains a data set)
- table (a set of records)
- module (a set of tables)
- application (a set of modules)

**Comment by fernandomalmeida** [comment]
What about an enum type? Maybe it is not so 'minimal', but can be useful (like a string, but with some constraints)

**Comment by crgimenes** [comment]
I don't think enum is for now, we need to define what is "essential" first, but I think the idea is interesting, open an issue for the enum to discuss how to implement it.

For now it's easy because the types I proposed are either mapable directly to Go or just structures.

**Comment by fernandomalmeida** [comment]
I Agree, about open an issue for the enum, I'll wait until there is some implementation so we'll have something to discuss about.
